snippets/linear_regression.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
+
+def compCostFunction(estim_y, true_y):
+    E = estim_y - true_y
+    C = (1 / 2 * m) * np.sum(E ** 2)
```

Comment:
where does the variable m come from here?

Reply:
Yes, global variables are terrible. Please move all the code defining variables into a function.